internal/impl/devbox.go

```diff
@@ -739,11 +758,13 @@ func (d *Devbox) computeNixEnv(setFullPath bool) (map[string]string, error) {
 		env[k] = v
 	}
 
-	// TODO: add shell-specific vars, including:
-	// - NIXPKGS_ALLOW_UNFREE=1 (not needed in run because we don't expect nix calls there)
-	// - __ETC_PROFILE_NIX_SOURCED=1 (not needed in run because we don't expect rc files to try to load nix profiles)
-	// - HISTFILE (not needed in run because it's non-interactive)
```

Comment:
would it be good to maintain somewhere a list of env vars like HISTFILE that are deliberately omitted from `devbox run` but included in `devbox shell`?

Reply:
maybe in LeakedVarsForShell with a value of `false` ?

Reply:
Well, I think looking at shellrc will be good enough for now (there's no other place in which the env can be changed)